src/lib/mgba/websocket-client.ts

```diff
@@ -0,0 +1,554 @@
+
+    const response = await this.eval(luaCode)
+    if (response.error) {
+      throw new Error(`Failed to bulk write ${data.length} bytes at 0x${address.toString(16)}: ${response.error}`)
+    }
+  }
+
+  /**
+   * Write multiple bytes using chunked approach for very large writes
+   */
+  async writeBytesChunked (address: number, data: Uint8Array, chunkSize = MEMORY_CONSTANTS.DEFAULT_CHUNK_SIZE): Promise<void> {
+    for (let offset = 0; offset < data.length; offset += chunkSize) {
+      const chunkEnd = Math.min(offset + chunkSize, data.length)
+      const chunk = data.slice(offset, chunkEnd)
+      await this.writeBytesBulk(address + offset, chunk)
+    }
+  }
+
+  /**
+   * Get the current game title to check compatibility
+   */
+  async getGameTitle (): Promise<string> {
+    const response = await this.eval('emu:getGameTitle()')
+    if (response.error) {
+      throw new Error(`Failed to get game title: ${response.error}`)
+    }
+    return response.result as string
+  }
+
+  /**
+   * Configure shared buffer settings
+   */
+  configureSharedBuffer (config: Partial<SharedBufferConfig>): void {
+    this.sharedBufferConfig = { ...this.sharedBufferConfig, ...config }
+  }
+
+  /**
+   * Preload commonly used memory regions into cache
+   */
+  async preloadSharedBuffers (): Promise<void> {
+    console.log('🔄 Preloading shared memory buffers...')
+
+    for (const region of this.sharedBufferConfig.preloadRegions) {
+      try {
+        await this.getSharedBuffer(region.address, region.size)
+        console.log(`✅ Preloaded region 0x${region.address.toString(16)} (${region.size} bytes)`)
+      } catch (error) {
+        console.warn(`⚠️  Failed to preload region 0x${region.address.toString(16)}: ${String(error)}`)
+      }
+    }
+  }
+
+  /**
+   * Get data from shared buffer (with caching)
+   * This method provides ultra-fast access to frequently used memory regions
+   */
+  async getSharedBuffer (address: number, size: number, useCache = true): Promise<Uint8Array> {
+    const cacheKey = `${address}-${size}`
+    const now = Date.now()
+
+    // Check cache first
+    if (useCache) {
+      const cached = this.memoryCache.get(cacheKey)
+      if (cached && !cached.dirty && (now - cached.lastUpdated) < this.sharedBufferConfig.cacheTimeout) {
+        return cached.data
+      }
+    }
+
+    // Read from memory using optimized method
+    const data = await this.readBytes(address, size)
+
+    // Cache the result
+    if (useCache) {
+      this.memoryCache.set(cacheKey, {
+        address,
+        size,
+        data: new Uint8Array(data), // Make a copy
+        lastUpdated: now,
+        dirty: false,
+      })
+
+      // Cleanup cache periodically (every 10 accesses) to avoid overhead
+      this.cacheAccessCount++
+      if (this.cacheAccessCount % 10 === 0) {
+        this.cleanupCache()
+      }
+    }
+
+    return data
+  }
```

Comment:
is there a lib that can handle best practice caching for us?
lets think about if that would improve our code maintainability